### ISSUE TITLE
Make ListPart reusable

### ISSUE THREAD

**Issue by giannik** [OrchardCMS/OrchardCore#3469]
When rendering multiple listparts set as reusable then only the first listpart is renderered multiple times.

To Repro: 
Use latest dev branch and run blog recipe.
From admin set the listpart  property Reusable =true.
Create a new content type (eg CustomBlog) and add 2 named listparts.

- Name first one "Articles" and set Contained Content Type to Article
- Name second one "BlogPosts"  and set Contained Content Type to BlogPost

Create a new content item of type CustomBlog  and save.

**Error** : The first listpart Articles is rendered twice (see screenshot below). 
**Expected behavior** : Each list part Articles and BlogPosts should be rendered individually.

![image](https://user-images.githubusercontent.com/1516981/56091202-3fdbbb80-5eb4-11e9-89fc-f4af6ee03dd4.png)

**Comment by sebastienros** [comment]
If you have to set it to true, then it means it was not marked as reusable, otherwise it would be. This setting should only be used for dynamically created parts.

Now I will also check if there is a reason we could not reuse it.

**Comment by giannik** [comment]
Thanks.
I can remember doing the same a few months back with listpart and it was working.
Has something changed on how/when to set a part as reusable ?
Giannis

**Comment by jptissot** [comment]
I would be interested to be able to use multiple Lists per ContentType as well. We are currently using the ContentPicker on the Child item but loose the Feed

**Comment by deanmarcussen** [comment]
I checked into this yesterday, as it might have been a useful feature in an upcoming project.

The reason it doesn't work (and I don't understand how it could ever have worked) is that the `ContainedPart.ListContentItemId` refers only to the parent content item id.

In order to make it reusable it would need to be prefixed with the Named List Part

So it would be `MyNamedListPart-ListContentItemId`, `FooPart-x523g436fgdf6456`

I tried doing this, and it works, but it looks very strange in the UI (as shown above) to have two lists showing.

So not putting a PR in, as it would need top many UI changes.

I think it would have to then have multiple links in the parent content item to each list-item

**Comment by giannik** [comment]
I forgot this issue, but the issue is this line of code in 

https://github.com/OrchardCMS/OrchardCore/blob/691ec3ab1467f3ec700e7b84ee4113da0b563ae4/src/OrchardCore.Modules/OrchardCore.Lists/Drivers/ListPartDisplayDriver.cs#L216

Just replace with this line : 
```
 var contentTypePartDefinition = contentTypeDefinition.Parts.FirstOrDefault(x => String.Equals(x.Name, partName, StringComparison.Ordinal));
``` 

annd pass the partname from the display method in the driver


```
 public override IDisplayResult Display(ListPart listPart, BuildPartDisplayContext context)
        {    
		        var settings = GetSettings(part,context.TypePartDefinition.Name);
				
     	}
```


Sorry, forgot  to submit a PR.

**Comment by deanmarcussen** [comment]
> I forgot this issue, but the issue is this line of code in

Or do that :) Seems simpler (but I think maybe it will still have problems, when it queries)

**Comment by giannik** [comment]
Yes , you are right . This is only half the solution. You still need to add an additional filter to the query that retrieves the child content items for that specific list part only.  Otherwise both list parts will show the same content. 
It worked for me a while back because I copied the list part module in a custom module and added the extra field to distinguish the part it belongs to.

**Comment by sebastienros** [comment]
Create a Content Type that has a ListPart (type B), Create a content type that has a Bag Part containing instances of type B. Would that work ?
